src/cljam/io/bam/decoder.clj

```diff
@@ -129,9 +156,14 @@
          seq         (decode-seq (lsb/read-bytes buffer (quot (inc l-seq) 2)) l-seq)
          qual        (decode-qual (lsb/read-bytes buffer l-seq))
          rest        (lsb/read-bytes buffer (options-size (alength ^bytes (:data block)) l-read-name n-cigar-op l-seq))
-         options     (decode-options rest)]
+         options     (decode-options rest)
+         [cigar* options*]
+         (if (and (contains? (first options) :CG)
+                  (first-cigar-operation-clips-entire-read? cigar l-seq))
```

Comment:
The order of this "and" conditionals looks good according to the following bench run.
https://github.com/chrovis/cljam/blob/5d1b98097197136a48171198803ed9233c72d465/bench/cljam/io/bam/decoder_bench.clj#L7-L21

Reversing the order of this "and" conditionals and executing the bench yields the following results.
These results are clearly worse.

```clojure
decode-bam-alignment-short-bench (decoder_bench.clj:7)

  ["test-resources/bam/test.bam"]
  time: 70.922426 µs, sd: 0.001001 ns

  ["test-resources/bam/medium.bam"]
  time: 37.774059 ms, sd: 210.986158 ns

  ["test-resources/bam/codec_benchmark.bam"]
  time: 112.478356 ms, sd: 3.596706 µs

decode-bam-alignment-long-bench (decoder_bench.clj:16)

  ["/Users/tmatsuda/clj/cljam_original/cljam/.cavia/large.bam"]
  time: 27.047076 sec, sd: 122.449742 ms
```